common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAccessToken.java

```diff
@@ -41,22 +41,22 @@
     public AzureActiveDirectoryAccessToken(
             @NonNull final AzureActiveDirectoryTokenResponse response) {
         super(response);
-        mExpiresOn = response.getExpiresOn();
-        mExtendedExpiresOn = response.getExtExpiresOn();
+        mExpiresOn = response.getExpiresOn() == null ? null : new Date(response.getExpiresOn().getTime());
+        mExtendedExpiresOn = response.getExtExpiresOn() == null ? null : new Date(response.getExtExpiresOn().getTime());
```

Comment:
This is where it would be convenient to move to java.time.  I believe that Instant instances are immutable, so can be safely returned.

Reply:
i'll create a card to track this. we can do once we go full java 8.